packages/flutter/lib/src/rendering/sliver_block.dart

```diff
@@ -497,6 +498,7 @@ abstract class RenderSliverBlock extends RenderSliver
       scrollExtent: estimatedTotalExtent,
       paintExtent: paintedExtent,
       maxPaintExtent: estimatedTotalExtent,
+      hasVisualOverflow: endScrollOffset > targetEndScrollOffset || startScrollOffset < constraints.scrollOffset,
```

Comment:
I don't understand the second term of this or.
Surely we have overflow any time constraints.scrollOffset is greater than 0.0? (Unless we happen to be really lucky and be exactly on a child boundary at our leading edge, but it seems like turning clipping on and off if we happen to hit that boundary is going to be pointless if not extra expensive.)